app/src/main/java/com/spoony/spoony/presentation/register/navigation/RegisterNavigation.kt

```diff
@@ -15,7 +17,12 @@ import com.spoony.spoony.presentation.register.RegisterViewModel
 import kotlinx.serialization.Serializable
 
 fun NavController.navigateToRegister(
-    navOptions: NavOptions? = null
+    navOptions: NavOptions = navOptions {
+        popUpTo(MainTab.MAP) {
+            inclusive = true
+        }
+        launchSingleTop = true
+    }
```

Comment:
```suggestion
    navOptions: NavOptions = navOptions {
        popUpTo(Register) {
            inclusive = false
        }
        launchSingleTop = true
    }
```

P3: 등록뷰로 이동하는 화면이 Map만 있는 것은 아니니까 popUpTo(Register)로 하고 inclusive를 false로 해주는게 더 맞지 않을까 싶은데 어떤가요??

Reply:
```suggestion
    navOptions: NavOptions? = null
```
P3: 그리고 MainNavigator에서 이미 navOpotions를 주고 있기 때문에 여기에서는 navOptions는 없어도 될 것 같아요!!

Reply:
일단 먼저 첫번째 방식은 기대하는 동작과 다르게 될 것 같아요. 지금 `navigateToRegister`의 시나리오 하나를 가정해보자면, Map -> Explore -> Map -> Register로 이동 시에 불러오는 함수인데, 이렇게 하면 Register는 아직 스택에 없는 상태이므로 popUpTo 동작이 아무 효과가 없게 된다고 예상이 됩니다. :0

때문에 저희가 의도했던 모든 백스택 제거를 하려면 다음처럼 수정해볼 수 있을 것 같아요
```kotlin
fun NavController.navigateToRegister(
    navOptions: NavOptions = navOptions {
        popUpTo(0) { inclusive = true }  // 모든 백스택을 완전히 제거
        // 또는 아래도 같은 동작입니다.
        // popUpTo(navController.graph.startDestinationId) { inclusive = true }
        launchSingleTop = true 
    }
) {
    navigate(Register, navOptions)
}
```
이렇게 하면 어떤 경로로 Register로 이동하든 이전 백스택이 모두 제거되어 Register에서 뒤로가기 시 앱이 종료됩니다.

Reply:
두번째로, MainNavigator에서 이미 navOptions를 주고 있습니다! 따라서 기본값으로 지정하지 않아도 될 것 같은데, 다만 이런 생각이 들었습니다. 

기능명세서에서 Register 화면에서 뒤로가기 시 앱이 종료되어야 한다는 기본 동작이 명시되어 있는데요, 이를 보장하기 위해 기본값으로 남겨두는 것은 어떨까요? 

물론 지금은 Register로 이동하는 호출부가 두 군데뿐이라 기본값 없이 각 호출부에서 필요한 navOptions를 직접 지정하는 것도 충분해 보이기도 하고.. 여러 의견을 들어보고 싶습니다 :)

Reply:
@chattymin @Roel4990 @Hyobeen-Park

Reply:
헉스 맞네요 popUpTo(0)으로 하는게 맞네요ㅎㅎ 예리했다!!

그리고 두 번째 이슈에 대해서는 현재 기능명세서 상 Register 화면에서 뒤로가기 시 앱은 무조건 종료되기 때문에 기본값을 넣어주는 것도 정말 좋은 생각입니다!! 다만 그렇다면 저희는 화면을 이동해도 state가 유지되도록 구현했기 때문에 saveState / restoreState도 함께 해주어야 기본값을 지정했을 때 저희가 의도한대로 될 것 같습니다

또한 mainNavigator에 스택을 없애기 위해 navOption을 전달하고 있는데 이 부분들은 불필요하지 않을까 하는 생각이 드네요!! 스택관리를 위한 navOptions를 기본값으로 지정했는데 mainNavigator에서  navOptions를 직접 지정해주면 사실 같은 navOptions를 두 번 지정하는 것이라는 생각이 들어서요..!! 이 부분에 대해서는 어떻게 생각하시는지 의견이 궁금합니다!

Reply:
1. `popUpTo(0)` 죠씁니다~~
2. 말씀해주신 것 처럼 Register화면의 기능 명세상 뒤로가기를 누르면 화면이 종료되어야 한다는 것이 고정이라면, 효빈님 말씀처럼 기본값을 설정하고 navOption을 넘겨주지 않는 방향으로 가는게 좋아보입니다. 매번 넣어주기보다는 기본값으로 있는게 더 적합해 보이네요

Reply:
위의 대화 이후 생각해보니, Register 화면으로의 네비게이션을 extension 함수에서 기본값으로 강제하기보다는 호출하는 쪽에서 제어하도록 하는 게 더 좋을 것 같았습니다.

```kotlin
fun NavController.navigateToRegister(
    navOptions: NavOptions? = null
) {
    navigate(Register, navOptions)
}
```
이렇게 수정한 이유는 나중에 Register 화면의 네비게이션 동작을 수정해야 할 때도 호출하는 쪽만 수정하면 되니 유지보수도 더 쉬워질 것 같습니다.

물론 Register 화면에서는 뒤로가기 시 앱이 종료되어야 한다는 요구사항이 있지만, 이건 각 호출부에서 상황에 맞게 처리하는 게 더 명확할 것 같네요.